src/date-picker/date-picker.jsx

```diff
@@ -177,7 +177,6 @@ const DatePicker = React.createClass({
 
   getInitialState() {
     return {
-      date: this._isControlled() ? this._getControlledDate() : this.props.defaultDate,
       dialogDate: new Date(),
```

Comment:
U wanted me to remove the dialogDate? But even removing the earlier statement didn't make.any difference.